completion_time_analysis.py

```diff
@@ -0,0 +1,237 @@
+"""
+Feature 2: Issue Completion Time Analysis (Focused)
+
+- Computes time-to-close for CLOSED issues only (no fallback).
+- ONE chart: Monthly median completion time (overall + top 3 labels by closed count).
+- Short summary: count, median, mean, P90; fastest/slowest labels if sample >= 3.
+"""
+
+from typing import List, Dict, Optional, Any
+from datetime import datetime, timezone
+
+import matplotlib.pyplot as plt
+import pandas as pd
+
+from data_loader import DataLoader
+from model import Issue, State
+import config
+
+
+# ----------------------------- helpers ------------------------------------ #
+
+def _to_utc_dt(x: Any) -> Optional[datetime]:
+    if x is None:
+        return None
+    if isinstance(x, datetime):
+        return x if x.tzinfo else x.replace(tzinfo=timezone.utc)
+    ts = pd.to_datetime(str(x), utc=True, errors="coerce")
+    if pd.isna(ts):
+        return None
+    return ts.to_pydatetime()
+
+def _created_at(issue: Issue) -> Optional[datetime]:
+    """Get created date from issue."""
+    return _to_utc_dt(issue.created_date) if issue.created_date else None
+
+def _updated_at(issue: Issue) -> Optional[datetime]:
+    """Get updated date from issue."""
+    return _to_utc_dt(issue.updated_date) if issue.updated_date else None
+
+def _closed_at_from_events(issue: Issue) -> Optional[datetime]:
+    """Find closed date from events."""
+    cands = []
+    for e in issue.events:
+        if e.event_type and e.event_type.lower() == "closed" and e.event_date:
+            ts = _to_utc_dt(e.event_date)
+            if ts:
+                cands.append(ts)
+    return max(cands) if cands else None
+
+def _closed_at(issue: Issue) -> Optional[datetime]:
+    """Get closed date from issue events, or fallback to updated_date if closed."""
+    ts = _closed_at_from_events(issue)
+    if ts:
+        return ts
+    # Fallback: if issue is closed, use updated_date
+    if issue.state == State.closed:
+        return _updated_at(issue)
+    return None
+
+def _labels(issue: Issue) -> List[str]:
+    """Get labels from issue."""
+    return issue.labels if issue.labels else ["unlabeled"]
+
+def _url(issue: Issue) -> Optional[str]:
+    """Get URL from issue."""
+    return issue.url if issue.url else f"https://github.com/python-poetry/poetry/issues/{issue.number}" if issue.number >= 0 else None
```

Comment:
Are there instances missing urls or labels? If so, then these are helpful. If not, then these aren't necessary